src/bin/chessmarkable/canvas.rs

```diff
@@ -58,6 +59,63 @@ impl<'a> Canvas<'a> {
         self.framebuffer_mut().wait_refresh_complete(update_marker);
     }
 
+    //Long text with draw_text layers on top of each other ending up in garbled output
+    //This is a quick hack which tries to prevent word clipping
+    //I've found that the remarkable can do about a 100 characters at 35.0 font size
+    //if you're looking for a default that fits the whole screen
+    pub fn draw_multi_line_text(&mut self, x_pos: Option<i32>, y_pos: i32, text: &str, max_chars_per_line: usize, size: f32) -> mxcfb_rect {
+        if text.len() > 0 {
+            let result = text.chars().enumerate().fold(vec![], |mut acc, p| {
+                if p.1 == ' ' {
+                    acc.push(p.0);
+                    acc
+                } else {
+                    acc
+                }
+            });
+            let mut vec_of_text = Vec::new();
+            let mut peekable = text.chars().peekable();
+            let mut last_text_height = 0;
+            let mut last_text_y = y_pos;
+            let mut chars_taken_so_far = 0;
+            let total_iterations = text.chars().count() / max_chars_per_line;
+            let mut current_iteration = 1;
+            while peekable.peek().is_some() {
+                let mut chars_to_take: usize = max_chars_per_line;
+                if current_iteration > total_iterations {
+                    chars_to_take = (text.len() + 1) - chars_taken_so_far;
+                } else {
+                    let option_chars_to_take = result.iter().rev().find(|char| **char < chars_taken_so_far + max_chars_per_line);
```

Comment:
If performance is a big issue, could optimise this with a binary search, but I was lazy to implement it, do you know if there are any out of the box solutions in Rust ?

Reply:
I don't think that this should be a big concern. The game is already not that fast, since I do compositing images over the framebuffer on a pixel by pixel basis (didn't find a better solution for the rM).

So doing char by char, should work. Didn't take a too close look, but maybe "dry running" each word/character could work to determine whether a potential part would go out of bounds. That could be used to split the text up properly.

I think, I used the "text dry run" method of libremarkable to center text in "buttons".

Reply:
Not sure myself, whether rust has something for binary search. Probably some sort of sorting libary.

Reply:
Yep, I think this may not be a bottleneck, so probably isn't a performance concern. Just thought I'd bring it up in case you want to optimize some parts of the code later.